core/execution/market.go

```diff
@@ -1478,6 +1419,9 @@ func (m *Market) submitValidatedOrder(ctx context.Context, order *types.Order) (
 		if isPegged {
 			m.peggedOrders.Park(order)
 		}
+		// parking the order, needs to unregister it first
+		_ = m.position.UnregisterOrder(ctx, order)
+		m.broker.Send(events.NewOrderEvent(ctx, order))
 		return &types.OrderConfirmation{Order: order}, nil, nil
```

Comment:
@EVODelavega @peterbarrow Just to confirm, when reading the code I realised that we may not have been unregistering position in this specific case of submit order. Would be good to have another pair of eyes there.

Reply:
Perhaps, that being said, having the `position.UnregisterOrder` call outside of the `if isPegged` does imply that we'd be calling this for both pegged orders and LPs, when in auction. I'm not 100% on whether we need it for both (although I'd need to take a bit more time to go through all possible scenarios

Reply:
hum good point. we should look into that yea.

Reply:
Alright, just looked  and that's be OK for pegged orders at least.

Reply:
Also looked at the liquidity orders, and that should work well for them as well. but please look on your side. I think that was actually really a bug before.